resources/lib/videoplayer.py

```diff
@@ -42,6 +42,9 @@ def __init__(self, args: Args, api: API):
         self._api = api
 
         self._stream_data: VideoPlayerStreamData | None = None
+        # @todo: what about movies and other future content types?
+        self._episode_data: EpisodeData | None = None
```

Comment:
What about movies?

Reply:
Hm I already thought of creating an Intermediate class for items that are playable, like the Episodes and Movies (and later others). Something like

```python
class PlayableItem(ListableItem):
    ...
   
class EpisodeData(PlayableItem):
    ...
   
class MovieData(PlayableItem):
    ...
```

That way those can be identified rather easily with something like `isinstance(item, PlayableItem)` whenever needed and you can define the type more generic in your code:
```python
self._media_data = PlayableItem | None = None  # no idea for a good property name right now :)
```

I guess it will hold the data fetched from the objects endpoint and that can be implemented rather agnostic of whether it is an Episode or an Movie - just load the data and pass it to `view.get_listables_from_response` to retrieve the DTO.

The whole stuff could be added to `VideoPlayerStreamData` for the new property and to `VideoStream` for the logic to retrieve that data.

I actually have a branch where I have added fetching the subtitles, skip events and playheads to these already and fetch them in parallel threads in an attempt to speed up the fetching - right now we have the issue that the API is called successively, one call after another, which delays things quite a bit if we want to poll multiple endpoints. There are more optimizations in my branch, but I have not merged it yet, as, while it works good for me, it probably isn't the best way as the requests library isn't meant for async operations and I'm sure it isn't thread-safe. I've then made another branch based upon that and rewrote things to use async operations based on `aiohttp`, only to find out after changing everything, that it doesn't exist in Kodi. :|

Anyways, fetching the object data could be easily added - and was planned - to my existing code inside `VideoStream` and `VideoStreamData`. I could merge that first branch I've mentioned to the dev branch, if you want to take a look.